App engine flex deployed

SimonBartonPSU opened the pull request:
Okey dokey, this appears to be working in my test GCP project. To be clear `10.4.48.3` is the Private IP addr of the DB instance cg-prod in GCP.

Things to note:

1. os.getenv() lines **ARE NOT WORKING** as is. I had to manually set the secret key and db password in my local copy of settings.py to get this deploy to work. I did not want to upload our secrets to GitHub again, so I left that out of the code.

 If you don't change manually, you will get a DJANGO_SECRET_KEY not set error or can't connect to DB error. I, for the life of me, could not figure out how to pass these two values through Dockerfile.... I tried many times. Whomever deploys will just have to set those vars locally, and we can fix this jenkyness in a future issue...

Also, remember that this Postgres DB has a new, secure password. Posted in Slack yesterday.

2. The `if os.getenv('GAE_APPLICATION', None):` trick does not work in App Engine flex, so I scrapped it. One database config.

3. Deployent should be performed with `gcloud beta app deploy` since we are now using beta features. Once deployed, one could watch log stream with `gcloud app logs tail -s default`.

4. Also once deployed, I would recommend running `./cloud_sql_proxy -instances=cellular-virtue-277000:us-central1:cg-prod=tcp:5432` then manually modifying DB hostname to 127.0.0.1 temporarily in order to run `python manage.py migrate` if need be.

After that I think everything would be good to go. Turned off SSL redirect because that was causing an issue as well. Honestly not sure how much that affects the project.

Made a slight adjustment to models as well, because I was getting warning that those `null`s were doing nothing when I migrated.

Not sure it fully resolves, but finishes a good chunk of #191.

SimonBartonPSU committed:
App engine flex deployment - #191

Found a way to use app engine flex, so that  our app engine is technically a VM instance running a container specified with our Docker file. Database is converted to private IP.

SimonBartonPSU committed:
Get deploy working sans ENV vars

Update DB settings

SimonBartonPSU committed:
Fix indentation

SimonBartonPSU commented:
For proof these config files lead to successful deploy vist http://ace-mote-270703.wl.r.appspot.com/postings/

(Obviously code has been changed back for our project DB instance, IP addr, etc, but it did work).

SimonBartonPSU commented:
One last thing: if you're looking to add additional config to environment, we can add more `RUN <some command here or something>` lines to Dockerfile. Or any Dockerfile [instructions](https://docs.docker.com/engine/reference/builder/) really.

EHaake commented:
Looks good, we'll need to try it out but looks really thorough with great documentation. Nice work

SimonBartonPSU committed:
foo

SimonBartonPSU committed:
Cleanin up

SimonBartonPSU committed:
Initial GIS deploys 7/16/20

Changed psycopg2 in requirements to psycopg2-binary.
Switched to Docker's python3.8 image. Added a couple more run commands
in Dockerfile.

SimonBartonPSU commented:
Deploy was successful with Sean and Erik today. [This ](http://ace-mote-270703.wl.r.appspot.com/auth/)endpoint is hosting app engine instance with the Dockerfile and app.yaml specified on branch app-engine-flex. (Contains postGIS dependencies, woot).

seansisson reviewed (approved):
Looks good, confirmed working as of yesterday
